userspace/falco/yaml_helper.h

```diff
@@ -72,6 +72,42 @@ class yaml_helper
 		get_node(node, key);
 		if(node.IsDefined())
 		{
+			// Check if the value is an environment variable reference
+			std::string value = node.as<std::string>();
+			if(!value.empty() && value[0] == '$')
+			{
+				std::string env_var;
+				if(value.size() > 1 && value[1] == '{' && value[value.size() - 1] == '}')
```

Comment:
what would be the value to resolve both `${VAR}` and `$VAR` format? If there is no difference I would go with just one possible way of specifying the env variable in the config -> `${VAR}` 
WDYT?

Reply:
I have removed the logic to process both ${VAR} and $VAR in favour of using just ${VAR} this conflicts with some of your test suggestions which I have updated accordingly.